### ISSUE TITLE
SpatialiteOnQt : "Unable to execute statement" "no such function : load_extension"

### ISSUE THREAD

**Issue by am2222** [eyllanesc/stackoverflow#2]
Hi,
I tried to use your provided example in stackoverflow but I faced this error.

`"Unable to execute statement" "no such function : load_extension"`
Is there any solutions for that?
thanks

**Comment by eyllanesc** [comment]
What is your OS, what version of sqlite and Qt do you have?

Edwin Christian Yllanes Cucho
Electronic Engineering Student
Universidad Nacional de Ingeniería, Lima, PERU
cel phone: (+51) 962 856 190
e-mail: e.yllanescucho@gmail.com


 Antes de imprimir este mensaje, asegúrese que sea necesario. El medio
ambiente está en nuestras manos.

2017-12-02 14:26 GMT-05:00 Majid Hojati <notifications@github.com>:

**Comment by am2222** [comment]
@eyllanesc  I am on Linux unbuntu and QT 5.9.1 and SQLite version 2.8.17

**Comment by eyllanesc** [comment]
You have installed the spatialite development package: libspatialite-dev?

**Comment by am2222** [comment]
@eyllanesc  yes I can call ` SELECT load_extension('mod_spatialite')` directly from any sqlite db manager and it works. Do I need recompile QT's sqlplugin?

**Comment by eyllanesc** [comment]
I did not have the need to do so, reviewing several forums I have seen several suggest recompiling the sqlplugin adding the following: sqlite.pri:

`DEFINES += SQLITE_ENABLE_LOAD_EXTENSION SQLITE_ENABLE_RTREE 
SQLITE_OMIT_COMPLETE `
